### ISSUE TITLE
Implementation of WARP Loss or pairwise ranking losses

### ISSUE THREAD

**Issue by Jbiloki** [tensorflow/tensorflow#25444]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 1.12.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Looking to implement warp loss, attempting to implement but getting gradient errors based on ops used. This loss has been implemented elsewhere such as pytorch (https://medium.com/@gabrieltseng/intro-to-warp-loss-automatic-differentiation-and-pytorch-b6aa5083187a) but many models are already built using tensorflow.

**Will this change the current api? How?**
Yes, I am looking to implement new loss functions for learning to rank problems.

**Who will benefit with this feature?**
All users
**Any Other info.**

I have begun to implement this feature but run into gradient computation problems. I am looking for help implementing the loss function.

**Comment by alextp** [comment]
I think this belongs better in a downstream project. Specifcally, have you looked at tf ranking https://github.com/tensorflow/ranking ?
